test/framework/service.go

```diff
@@ -18,7 +20,22 @@ const (
 
 func (f *Framework) doRequest(endpoint, method string, body []byte, query map[string]string) (respBody []byte, code int, err error) {
 	var u *url.URL
-	if f.UseKubeProxyForReportingAPI {
+
+	if f.UseRouteForReportingAPI {
+		// query all routes for the metering route
+		meteringRoute, err := f.RouteClient.Routes(f.Namespace).Get("metering", metav1.GetOptions{})
+		if err != nil {
+			return nil, 0, err
+		}
+
+		// clean this up later - add protocol, path, etc.
+		// this evalutes to: https://metering-metering-tflannag.apps.metering-ci-3.origin-gce.dev.openshift.com/api/v1/reports/get?name=namespace-cpu-usage-runonce&namespace=metering-tflannag&format=json
+		urlReq := fmt.Sprintf("https://%s%s?name=%s&namespace=%s&format=%s", meteringRoute.Spec.Host, endpoint, query["name"], query["namespace"], query["format"])
```

Comment:
Maybe set the fields of `url.URL` directly instead of parsing a string.
Eg:
```
u = url.URL{
  Scheme: "https",
  Host: meteringRoute.Spec.Host,
  Path: endpoint,
}
```

No need to mess with `query`, as the queryParams get set down below https://github.com/operator-framework/operator-metering/pull/787/files#diff-d04c46492f473219243f718a664b8269R74